frontend/src/lib/ErrorServeice.ts

```diff
@@ -0,0 +1,223 @@
+import * as Sentry from '@sentry/react'
+
+import logger from 'utils/logger'
+import type { ErrorConfig } from 'lib/types'
+
+class ErrorService {
+  private readonly errorGroups: Map<string, ErrorConfig[]> = new Map([
+    [
+      'search',
+      [
+        {
+          code: 'ALGOLIA_SEARCH_FAILED',
+          title: 'Search Error',
+          message: 'Unable to complete search. Please try again.',
+          source: 'algolia',
+          action: 'retry',
+        },
+        {
+          code: 'ALGOLIA_CONFIG_ERROR',
+          title: 'Config Error',
+          message: 'Algolia keys not found.',
+          source: 'algolia',
+          action: 'retry',
+        },
```

Comment:
I'm not sure if we want to show this to user. It should never happen.

Reply:
In my first PR, I only handled 404, 500, and JavaScript errors. However, later we started fetching data directly from Algolia on the frontend. That’s why I added this part—to allow developers to see errors directly on the screen, not for users. I will remove this part, as well as the unnecessary elements like authentication and rate-limiting issues, which are not aligned with our current Nest.js project